core/storage/base_model/gae_models.py

```diff
@@ -1010,10 +1054,18 @@ def delete( # type: ignore[override]
             commit_cmds = [{
                 'cmd': self.CMD_DELETE_COMMIT
             }]
-
-            self._trusted_commit(
-                committer_id, self._COMMIT_TYPE_DELETE, commit_message,
-                commit_cmds)
+            models_to_put = cast(
```

Comment:
Why do you need to cast here? (Checking with @vojtechjelinek if this is expected, we typically try to avoid casting where possible.)

Ditto below.

Reply:
Yeah it is needed here.

Reply:
I had removed this and the job is working fine, looks like the cast is not needed.